File: Java/NetBeans2/thiguaky/src/thiguaky/SinhVien.java
package thiguaky;

public class SinhVien {
    
    private String hoTen;
    private String nganhHoc;
    private int soTinChi;

    public SinhVien() {
    }

    public SinhVien(String hoTen, String nganhHoc, int soTinChi) {
        this.hoTen = hoTen;
        this.nganhHoc = nganhHoc;
        this.soTinChi = soTinChi;
    }

    public double hocPhi() {
        switch (nganhHoc) {
            case "102" -> {
                return 200000 * soTinChi;
            }
            case "103" -> {
                return 220000 * soTinChi;
            }
            default -> {
                return 0;
            }
        }
    }
    
    public void hienThi() {
        System.out.println("Ho ten: " + hoTen);
        System.out.println("Nganh hoc: " + nganhHoc);
        System.out.println("So tin chi: " + soTinChi);
        System.out.println("Hoc phi: " + hocPhi());
    }
}

File: C#/DiemHP/DiemHP/SinhVien.cs
namespace DiemHP
{
    public class SinhVien
    {
        private string hoTen;
        private DiemHP hp1;
        private DiemHP hp2;

        public SinhVien(string hoTen, DiemHP hp1, DiemHP hp2)
        {
            this.hoTen = hoTen;
            this.hp1 = hp1;
            this.hp2 = hp2;
        }

        public void hienThi()
        {
            Console.WriteLine($"Ho ten: {hoTen}");
            hp1.hienThi();
            hp2.hienThi();
        }

        public float tinhDTB()
        {
            return (hp1.tinhDiem() * hp1.SoTinChi + hp2.tinhDiem() * hp2.SoTinChi) / (hp1.SoTinChi + hp2.SoTinChi);
        }
    }
}
